singularity-eos/closure/kinetic_phasetransition_methods.hpp

```diff
@@ -0,0 +1,102 @@
+//------------------------------------------------------------------------------
+// © 2024. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract 89233218CNA000001
+// for Los Alamos National Laboratory (LANL), which is operated by Triad
+// National Security, LLC for the U.S.  Department of Energy/National
+// Nuclear Security Administration. All rights in the program are
+// reserved by Triad National Security, LLC, and the U.S. Department of
+// Energy/National Nuclear Security Administration. The Government is
+// granted for itself and others acting on its behalf a nonexclusive,
+// paid-up, irrevocable worldwide license in this material to reproduce,
+// prepare derivative works, distribute copies to the public, perform
+// publicly and display publicly, and to permit others to do so.
+//------------------------------------------------------------------------------
+
+#ifndef _SINGULARITY_EOS_CLOSURE_KINETIC_PHSETRANSITION_METHODS_
+#define _SINGULARITY_EOS_CLOSURE_KINETIC_PHSETRANSITION_METHODS_
+
+#include <ports-of-call/portability.hpp>
+#include <ports-of-call/portable_errors.hpp>
+#include <singularity-eos/base/fast-math/logs.hpp>
+#include <singularity-eos/base/robust_utils.hpp>
+#include <singularity-eos/eos/eos.hpp>
+
+#include <cmath>
+
+#ifdef SINGULARITY_USE_KOKKOSKERNELS
+#include <KokkosBatched_ApplyQ_Decl.hpp>
+#include <KokkosBatched_QR_Decl.hpp>
+#include <KokkosBatched_Trsv_Decl.hpp>
+#else
+#include <Eigen/Dense>
+#endif // SINGULARITY_USE_KOKKOSKERNELS
+
+namespace singularity {
+
+PORTABLE_INLINE_FUNCTION void SmallStepMFUpdate(const Real logdt, const int num_phases,
+                                                const Real *massfractions,
+                                                const int *gibbsorder, const Real *logRjk,
+                                                Real *dmfs, Real *newmfs) {
+
+  Real minmassfraction = 1.e-10;
```

Comment:
A couple of things:
1. Any compile time constants should be at least `const` or even better `constexpr`.
2. It appears that this constant is used in several files. I would propose moving this to `kinetic_phasetransition_util.hpp` and then define it as
```c++
constexpr Real KPT_MIN_MASS_FRACTION = 1.0e-10;
```
there. I'm using the convention that compile-time constants are all-caps.

This way we can define this parameter once and then use it in several files. If a host code needs this value for some reason, it's also exposed and easily queried. Further, if we want to expose it as a user-settable parameter in the future (maybe we don't... but who knows?), it's easier to do this if we've already extracted it from the places it is used.

Reply:
I second all of this. Also, 1e-10 is perfectly reasonable, but I'm curious why you chose that value in particular.

Reply:
Done